src/webots/core/WbStandardPaths.cpp

```diff
@@ -205,20 +216,18 @@ const QString &WbStandardPaths::webotsTmpPath() {
     }
 #endif
 
-    // remove the other Webots instances from the temp folder if needed
-    if (WbWebotsInstancesCounter::numberOfInstances() == 1) {
-      QDir tmpDir(webotsTmpPath);
-      tmpDir.cdUp();
-
-      tmpDir.setFilter(QDir::Dirs);
-      QStringList filters;
-      filters << "webots-*";
-      tmpDir.setNameFilters(filters);
-
-      QFileInfoList list = tmpDir.entryInfoList();
-      for (int i = 0; i < list.size(); ++i) {
-        QDir dirToRemove(list.at(i).absoluteFilePath());
-        dirToRemove.removeRecursively();
+    // cleanup old and unused tmp directories
+    const QString &tmpRoot = webotsTmpPath.left(webotsTmpPath.chopped(1).lastIndexOf('/'));
+    const QDir directory(tmpRoot);
```

Comment:
This part looks a bit too 'C' style to me :wink: 

```
    QDir directory(webotsTmpPath);
    directory.cdUp();
```

Seems simpler and clearer.